xilinx/common/cmake/arch_define.cmake

```diff
@@ -14,23 +41,29 @@ function(ADD_XC7_ARCH_DEFINE)
   set(DEFAULT_IOSTANDARD "LVCMOS33")
   set(DEFAULT_DRIVE 12)
 
-  set(ARCH ${ADD_XC7_ARCH_DEFINE_ARCH})
-  if("${ADD_XC7_ARCH_DEFINE_PRJXRAY_ARCH}" STREQUAL "")
-      set(PRJXRAY_ARCH "${ARCH}")
+  set(FAMILY ${ADD_XILINX_ARCH_DEFINE_FAMILY})
+  set(ARCH ${ADD_XILINX_ARCH_DEFINE_ARCH})
+  if("${ADD_XILINX_ARCH_DEFINE_PRJRAY_ARCH}" STREQUAL "")
+      set(PRJRAY_ARCH "${ARCH}")
   else()
-      set(PRJXRAY_ARCH "${ADD_XC7_ARCH_DEFINE_PRJXRAY_ARCH}")
+      set(PRJRAY_ARCH "${ADD_XILINX_ARCH_DEFINE_PRJRAY_ARCH}")
   endif()
-  set(PROTOTYPE_PART ${ADD_XC7_ARCH_DEFINE_PROTOTYPE_PART})
-  set(YOSYS_SYNTH_SCRIPT ${ADD_XC7_ARCH_DEFINE_YOSYS_SYNTH_SCRIPT})
-  set(YOSYS_CONV_SCRIPT ${ADD_XC7_ARCH_DEFINE_YOSYS_CONV_SCRIPT})
+  set(PRJRAY_DIR ${ADD_XILINX_ARCH_DEFINE_PRJRAY_DIR})
+  set(PRJRAY_DB_DIR ${ADD_XILINX_ARCH_DEFINE_PRJRAY_DB_DIR})
+  set(PROTOTYPE_PART ${ADD_XILINX_ARCH_DEFINE_PROTOTYPE_PART})
+  set(YOSYS_SYNTH_SCRIPT ${ADD_XILINX_ARCH_DEFINE_YOSYS_SYNTH_SCRIPT})
+  set(YOSYS_CONV_SCRIPT ${ADD_XILINX_ARCH_DEFINE_YOSYS_CONV_SCRIPT})
 
   define_arch(
     ARCH ${ARCH}
+    FAMILY ${FAMILY}
+    DOC_PRJ ${PRJRAY_DIR}
+    DOC_PRJ_DB ${PRJRAY_DB_DIR}
     PROTOTYPE_PART ${PROTOTYPE_PART}
     YOSYS_SYNTH_SCRIPT ${YOSYS_SYNTH_SCRIPT}
     YOSYS_CONV_SCRIPT ${YOSYS_CONV_SCRIPT}
     DEVICE_FULL_TEMPLATE \${DEVICE}-\${PACKAGE}
-    CELLS_SIM ${YOSYS_DATADIR}/xilinx/cells_sim.v ${symbiflow-arch-defs_SOURCE_DIR}/xc7/techmap/cells_sim.v
+    CELLS_SIM ${YOSYS_DATADIR}/xilinx/cells_sim.v ${symbiflow-arch-defs_SOURCE_DIR}/xilinx/common/techmap/cells_sim.v
```

Comment:
That will have to be specialized per `PRJRAY_ARCH`

Reply:
Right, done, moved all techmap stuff in `xc7` directory.